internal/rules/migrate.go

```diff
@@ -16,6 +18,64 @@ type Migrate struct {
 	Symlink bool
 }
 
+func MoveDirectory(source, dest string) error {
+	err := copy.Copy(source, dest)
+	if err != nil {
+		return err
+	}
+
+	err = os.RemoveAll(source)
+	if err != nil {
+		tui.Warn("Failed to remove original directory: %s", err)
+	}
+
+	return nil
+}
+
+// Adapted from https://gist.github.com/var23rav/23ae5d0d4d830aff886c3c970b8f6c6b
+func MoveFile(source, dest string) error {
+	inputFile, err := os.Open(source)
+	if err != nil {
+		return err
+	}
+
+	outputFile, err := os.Create(dest)
+	if err != nil {
+		inputFile.Close()
+		return err
+	}
+	defer outputFile.Close()
+
+	_, err = io.Copy(outputFile, inputFile)
+	inputFile.Close()
+	if err != nil {
+		return err
+	}
+
+	// The copy was successful, so now delete the original file
+	err = os.Remove(source)
+	if err != nil {
+		tui.Warn("Failed to remove original file: %s", err)
+	}
+
+	return nil
+}
+
+// Try to move file/directory with os.Rename and if that fails, do a copy + delete
+func Move(source, dest string) error {
+	err := os.Rename(source, dest)
+	if err == nil {
```

Comment:
I think it is safer to check the error details. It could be that the source file is not readable or there is not enough space on disk for the rename. In these cases copying the files won't work either.

Reply:
I've had a look at this and I can't see a nice way to detect whether this specific error has occurred or not as opposed to some other error in a cross-platform way :disappointed:. I feel like it doesn't matter too much if the code always falls back on trying to copy + delete in case of error because, as you say, if there's some other problem you'll see the error then anyway. It might be a more robust solution than some hack to try to check for this particular issue.